ReactiveCocoa/Swift/EventLogger.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  EventLogger.swift
+//  ReactiveCocoa
+//
+//  Created by Rui Peres on 30/04/2016.
+//  Copyright © 2016 GitHub. All rights reserved.
+//
+
+import Foundation
+
+public enum SignalLoggingEvent: String {
+	case Started, Next, Completed, Failed, Terminated, Disposed, Interrupted
+	
+	public static let allEvents: Set<SignalLoggingEvent> = {
+		return [.Next, .Completed, .Failed, .Terminated, .Disposed, .Interrupted]
+	}()
+}
+
+public enum SignalProducerLoggingEvent: String {
+	case Started, Next, Completed, Failed, Terminated, Disposed, Interrupted
+	
+	public static let allEvents: Set<SignalProducerLoggingEvent> = {
+		return [.Started, .Next, .Completed, .Failed, .Terminated, .Disposed, .Interrupted]
+	}()
+}
+
+private func printLog(identifier: String, event: String, fileName: String, functionName: String, lineNumber: Int) -> Void {
+	print("[\(identifier)] \(event) fileName: \(fileName), functionaName: \(functionName), lineNumber: \(lineNumber)")
+}
+
+public typealias EventLogger = (identifier: String, event: String, fileName: String, functionName: String, lineNumber: Int) -> Void
+
+extension SignalType {
+	/// Logs all events that the receiver sends.
+	/// By default, it will print to the standard output.
+	@warn_unused_result(message="Did you forget to call `observe` on the signal?")
+	public func logEvents(identifier: String = "", events: Set<SignalLoggingEvent> = SignalLoggingEvent.allEvents, fileName: String = #file, functionName: String = #function, lineNumber: Int = #line, logger: EventLogger = printLog) -> Signal<Value, Error> {
```

Comment:
What do you think about making identifier an explicit parameter name? That way callers have to say:

logEvents(identifier: "foo")

And not just

logEvents("foo")